pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -148,24 +180,32 @@ func ingest(
 
 		// Construct stream ingestion processor specs.
 		streamIngestionSpecs, streamIngestionFrontierSpec, err := distStreamIngestionPlanSpecs(
-			streamAddress, topology, sqlInstanceIDs, initialHighWater, ingestionJobID, streamID, oldTenantID, newTenantID)
+			streamAddress, topology, sqlInstanceIDs, progress.GetStreamIngest().StartTime,
+			ingestionJob.ID(), streamID, details.TenantID, details.NewTenantID)
```

Comment:
I don't see in this code how do we ever retry (i.e. hit the `for r := retry.Start(ro); r.Next(); {` line).  It seems that we bubble up every error encountered in this function.

distStreamIngestion for example could return an error for any number of legitimate reasons (node crashed; or perhaps restarted, etc).

I think part of this is code structure, as mentioned above:
have a resumer call a function `ingestWithRetry`; that function should be simple:
```
func ingestWithRetry() {
   for retry ... {
     err := ingest()
     // Handle error appropriately
     if retryableError {
        // reload job; log error, etc
     } else {
       // pause the job
    }
```

How we handle errors -- depends.  We could say all errors get retried, or perhaps we retry some of the errors.